app/controllers/metadata_controller.rb

```diff
@@ -2,13 +2,8 @@ class MetadataController < ApplicationController
   include MetadataHelper
 
   # Token auth needed for CLI uploads
-  skip_before_action :verify_authenticity_token, only: [:validate_csv_for_new_samples]
```

Comment:
What was the goal of this call? Why would it skip it? Is it done anywhere else?

Reply:
Thank you for catching this. I cherry picked that commit from a different branch and it was a test that I wasn't supposed to commit. (we have an issue in our git pre-commit script that commit all the lines even when they are not added to the commit, but this is a different discussion).

It looks like it is done in different places. If I understood it correct, this is because CLI doesn't send CSRF tokens, and therefore all endpoints that are invoked by CLI need to disable some of these constraints.

Actually I really don't like the idea we have these endpoints shared between CLI and WEB, because it compromises the security. We should either have separate endpoints, or have better support to these bearer tokens (that's something that I'm planning to do during auth0 migration), or maybe both.

This can be even a larger discussion. For me, ideally, we would only use bearer tokens, and would have a single page app, but this is not rails works by default (it uses cookies to store authentication tokens and renders pages as you click them, and for that reason it requires that CSRF tokens approach, which are usually limited to post operations). However, such change would be huge, not sure if we can fit all this extra work in our auth0 tasks. Would love to discuss more about that.